src/Adapter/Util/Entity/EntityNameDuplicator.php

```diff
@@ -0,0 +1,132 @@
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class EntityNameDuplicator
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var StringModifierInterface
+     */
+    private $stringModifier;
+
+    /**
+     * @var array
+     */
+    protected $languages;
+
+    /**
+     * @var array
+     */
+    protected $locales;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param StringModifierInterface $stringModifier
+     * @param array $languages
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        StringModifierInterface $stringModifier,
+        array $languages
+    ) {
+        $this->translator = $translator;
+        $this->stringModifier = $stringModifier;
+        $this->languages = $languages;
+    }
+
+    /**
+     * Adds a "copy" word to localized entity names
+     *
+     * @param array<int, string> $currentLocalizedNames Array with localized names, IdLang => Name
+     * @param int $maxLength max length of a new name, will be cut if it doesn't fit with the new "copy" word
+     *
+     * @return array<int, string>
+     */
+    public function getNewLocalizedNames(array $currentLocalizedNames, int $maxLength): array
+    {
+        if (empty($this->locales)) {
+            $this->buildLanguages();
+        }
+
+        $newLocalizedNames = [];
+        foreach ($currentLocalizedNames as $langId => $currentName) {
+            $langId = (int) $langId;
+
+            // Add copy word to the string
+            $namePattern = $this->translator->trans('copy of %s', [], 'Admin.Catalog.Feature', $this->locales[$langId]);
+            $newName = sprintf($namePattern, $currentName);
+
+            // Save and cut the end if it doesn't fit to the limit
+            $newLocalizedNames[$langId] = $this->stringModifier->cutEnd($newName, $maxLength);
+        }
+
+        return $newLocalizedNames;
+    }
+
+    /**
+     * Appends a "copy" to simple entity name
+     *
+     * @param string $currentName Current name of the entity
+     * @param int $maxLength max length of a new name, will be cut if it doesn't fit with the new "copy" word
+     * @param int $langId
+     *
+     * @return string
+     */
+    public function getNewName(string $currentName, int $maxLength, int $langId): string
+    {
+        if (empty($this->locales)) {
+            $this->buildLanguages();
+        }
+
+        // Add copy word to the string
+        $namePattern = $this->translator->trans('copy of %s', [], 'Admin.Catalog.Feature', $this->locales[$langId]);
+        $newName = sprintf($namePattern, $currentName);
+
+        // Cut the end if it doesn't fit to the limit and return
+        return $this->stringModifier->cutEnd($newName, $maxLength);
+    }
+
+    /**
+     * Builds the locale array
+     *
+     * @return void
+     */
+    private function buildLanguages(): void
```

Comment:
noticed just now, minor stuff, can change it during other fixes.
```suggestion
    private function buildLocales(): void
```